responsemanager/server.go

```diff
@@ -66,13 +82,18 @@ func (rm *ResponseManager) processUpdate(key responseKey, update gsmsg.GraphSync
 		return nil
 	})
 	if result.Err != nil {
+		response.span.RecordError(result.Err)
+		response.span.SetStatus(codes.Error, result.Err.Error())
 		delete(rm.inProgressResponses, key)
 		response.cancelFn()
+		response.span.End()
```

Comment:
do we have a problem with the defer statement above here? I think you're closing the parent span before the child span no?

Reply:
I had `defer`s on all of these `response.span.End()` calls but it was causing timing problems with the tests; removing them made all the problems disappear.

In terms of one span closing before the other, I don't think it actually matters all that much, the still retain their relationships and I think it's up to the visualisation tooling how they might deal with any minor discrepancies in end times but it'd be so close that it's probably not even going to be different in most cases.

But I'll try and revisit this close timing after I update to the new task end work you did and see how it goes.